File: sujeong/week3/조이 스틱.cpp
#include <algorithm>
#include <string>

using namespace std;

int solution(string name) {
    int n = name.size();
    int upDown = 0;
    int leftRight = n-1;
    
    for(int i = 0 ; i < n ; i++){
        upDown += min(name[i]-'A', 'Z'-name[i]+1);
        
        int ii = i+1;
        while(ii < n && name[ii] == 'A')
            ii++;
        
        int newLeftRight = i + n-ii + min(n-ii, i);
//        int newLeftRight = min(i*2 + n-ii, i + (n-ii)*2);

        leftRight = min(leftRight, newLeftRight);
    }
    
    return upDown + leftRight;
}

File: hyein/week3/조이스틱.py
def solution(name):
    answer = 0
    
    n = len(name)
    move = n - 1
    for idx, c in enumerate(name):
        answer += min(ord(c) - ord('A'), ord('Z') - ord(c) + 1)
        
        next_idx = idx + 1
        while next_idx < n and name[next_idx] == 'A':
            next_idx += 1
            
        distance = min(idx, n - next_idx)
        move = min(move, idx + n - next_idx + distance)

    return answer + move